clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -681,6 +682,12 @@ protected void checkLongRunningTaskState() throws InterruptedException {
             restoreProducerStatesForLeaderConsumption(partition);
             startConsumingAsLeader(partitionConsumptionState);
 
+            // Start tracking leader replication lag
+            getHeartbeatMonitoringService().updateLagMonitor(
```

Comment:
So this function is going to get called in a poll()'ing fashion.  Does that impact this?

Reply:
It will be called only once when it can be switched to leader, it won't be called in every check long running state